File: NetCore/CodeBattleNetLibrary/GameModels/Construction.cs
namespace CodeBattleNetLibrary.GameModels
{
    public class Construction : Point
    {
        public int Timer { get; private set;}
        public int Power { get; private set;}
        public Construction(int x, int y, int timer, int power) : base(x, y)
        {
            Timer = timer;
            Power = power;
        }
    }
}

File: java/src/main/java/clientlib/model/Construction.java
package clientlib.model;

import lombok.NoArgsConstructor;

@NoArgsConstructor
public class Construction extends Point {

    private int timer;
    private int power;

    public Construction(int x, int y, int timer, int power) {
        super(x, y);
        this.timer = timer;
        this.power = power;
    }

    public int getTimer() {
        return timer;
    }

    public void setTimer(int timer) {
        this.timer = timer;
    }

    public int getPower() {
        return power;
    }

    public void setPower(int power) {
        this.power = power;
    }

    public int decPowerAndGet() {
        power--;
        return power;
    }
}
